bindings_test.go

```diff
@@ -930,3 +930,137 @@ func TestFunctionParameters(t *testing.T) {
 		}
 	})
 }
+
+func TestVariousBindingModesForNonNullValues(t *testing.T) {
+	testcases := []struct {
+		testDesc  string
+		paramType string
+		input     interface{}
+	}{
+		{"textAndString", "text", "string"},
+		{"numberAndInteger", "number", 123},
+		{"floatAndFloat", "float", 123.01},
+		{"booleanAndBoolean", "boolean", true},
+		//{"dateAndTime", "date", time.Now()}, // does not bind in any binding mode
+		{"datetimeAndTime", "datetime", time.Now()},
+		//{"timeAndTime", "time", time.Now()}, // does not bind in any binding mode
+		{"timestampAndTime", "timestamp", time.Now()},
+		{"timestamp_ntzAndTime", "timestamp_ntz", time.Now()},
+		{"timestamp_ltzAndTime", "timestamp_ltz", time.Now()},
+		{"timestamp_tzAndTime", "timestamp_tz", time.Now()},
+	}
+
+	bindingModes := []struct {
+		desc      string
+		insert    string
+		query     string
+		transform func(interface{}) interface{}
+	}{
+		{
+			desc:      "?",
+			insert:    "INSERT INTO BINDING_MODES VALUES(?)",
+			query:     "SELECT * FROM BINDING_MODES WHERE param1 = ?",
+			transform: func(v interface{}) interface{} { return v },
+		},
+		{
+			desc:      ":1",
+			insert:    "INSERT INTO BINDING_MODES VALUES(:1)",
+			query:     "SELECT * FROM BINDING_MODES WHERE param1 = :1",
+			transform: func(v interface{}) interface{} { return v },
+		},
+		{
+			desc:      ":id",
+			insert:    "INSERT INTO BINDING_MODES VALUES(:param)",
+			query:     "SELECT * FROM BINDING_MODES WHERE param1 = :param",
+			transform: func(v interface{}) interface{} { return sql.Named("param", v) },
+		},
+	}
+
+	runTests(t, dsn, func(dbt *DBTest) {
+		for _, tc := range testcases {
+			for _, bindingMode := range bindingModes {
+				t.Run(tc.testDesc+" "+bindingMode.desc, func(t *testing.T) {
+					query := fmt.Sprintf(`CREATE OR REPLACE TABLE BINDING_MODES(param1 %v)`, tc.paramType)
```

Comment:
who drops that table after tests?

Reply:
The whole schema is dropped when tests are finished.

Reply:
No, in developer machine it's the same. See [teardown](https://github.com/snowflakedb/gosnowflake/blob/master/driver_test.go#L134) invoked after [all tests](https://github.com/snowflakedb/gosnowflake/blob/master/driver_test.go#L158).